### PR TITLE
afew: add module

### PR DESCRIPTION
**Pull request by bsima**
[Afew](https://github.com/afewmail/afew) is an initial tagging script for notmuch mail. This module can be used like so:

```nix
{
  programs.afew = {
    enable = true;
    extraConfig = builtins.readFile ./afew.ini;
  };
}
```

Where `afew.ini` is the config file for afew.

### CODE REVIEW

File: modules/programs/afew.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    extraConfig = mkOption {
+      type = types.lines;
+      default = ''
+        [SpamFilter]
+        [KillThreadsFilter]
+        [ListMailsFilter]
+        [ArchiveSentMailsFilter]
+        [InboxFilter]
+      '';
```
Comment (teto) [inline modules/programs/afew.nix]: Does this do anything by default ? you could add an "example" entry

**Comment by teto** [comment]
also you may want to add it to modules/misc/news.nix

**Comment by bsima** [comment]
Okay, news item and example added.

File: modules/programs/afew.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  configFile = ''
+    # Generated by Home Manager.
+    # See https://afew.readthedocs.io/en/latest/index.html
+
+    ${cfg.extraConfig}
+  '';
+in
+{
+  options.programs.afew = {
+    enable = mkOption {
```
Comment (rycee) [inline modules/programs/afew.nix]:
```nix
enable = mkEnableOption "the Afew initial tagging script for Notmuch";
```

**Comment by bsima** [comment]
Robert Helgesson <notifications@github.com> writes:

> rycee commented on this pull request.
>
>
>
>> +{ config, lib, pkgs, ... }:
> +
> +with lib;
> +
> +let
> +  cfg = config.programs.afew;
> +  configFile = ''
> +    # Generated by Home Manager.
> +    # See https://afew.readthedocs.io/en/latest/index.html
> +
> +    ${cfg.extraConfig}
> +  '';
> +in
> +{
> +  options.programs.afew = {
> +    enable = mkOption {
>
> ```nix
> enable = mkEnableOption "the Afew initial tagging script for Notmuch";
> ```

Ah, wasn't aware of this. Fixed in 125da0330d4e14fd7fb0c28dc3a8b1525fe3221a

**Comment by rycee** [comment]
Rebased to master in 061c7b633f66ff73b281f08991c6c0a541877e66. Thanks!

**Pull request closed without merging**